src/pharmpy/modeling/expressions.py

```diff
@@ -978,7 +976,20 @@ def _replace_trivial_redefinitions(model):
             d[s.symbol] = s.expression
         else:
             keep.append(s)
-    new = Statements(tuple(keep)).subs(d)
+
+    # Remove redundant assignemnt, for instance
+    # {KA : 1/MAT, K45: KA}
+    def update_value(temp_d, k):
+        if temp_d[k] in temp_d.keys() and temp_d[k] != k:
+            return update_value(temp_d, d[k])
+        else:
+            return temp_d[k]
+
+    clean_d = d.copy()
+    for k, v in d.items():
+        clean_d[k] = update_value(d, k)
+
+    new = Statements(tuple(keep)).subs(clean_d)
```

Comment:
Couldn't all of this be done simpler directly in the first for loop directly? 

isn't it simply to apply the subs dict to the rhs for each iteration? Or optionally perform a check similarly to what you did?

In any case it is a good catch!

Reply:
Optimized the function to perform the same substitution but within the first for-loop